test/test_autograd.py

```diff
@@ -4675,74 +4674,24 @@ def f(inp):
-
-            gradcheck(fn, torch.rand(10, dtype=torch.double).to_sparse().requires_grad_(True), check_sparse_nnz=True,
-                      check_batched_grad=False, fast_mode=fast_mode)
-            with self.assertRaisesRegex(RuntimeError, 'gradcheck expects all tensor inputs are dense'):
-                gradcheck(fn, torch.rand(10, dtype=torch.double).to_sparse().requires_grad_(True), check_sparse_nnz=False,
-                          check_batched_grad=False, fast_mode=fast_mode, masked=True)
-        check(fast_mode=True)
-        check(fast_mode=False)
-
-    def test_gradcheck_sparse_csr_input(self):
-        def check(fast_mode):
-            def fn(sparse_csr):
-                return torch.clone(sparse_csr).to_dense()
-
-            gradcheck(fn, torch.rand(2, 2, dtype=torch.double).to_sparse_csr().requires_grad_(True), check_sparse_nnz=True,
-                      check_batched_grad=False, fast_mode=fast_mode)
-
-            with self.assertRaisesRegex(RuntimeError, 'gradcheck expects all tensor inputs are dense'):
-                gradcheck(fn, torch.rand(2, 2, dtype=torch.double).to_sparse_csr().requires_grad_(True), check_sparse_nnz=False,
-                          check_batched_grad=False, fast_mode=fast_mode, masked=True)
-        check(fast_mode=True)
-        check(fast_mode=False)
-
-    def test_gradcheck_sparse_csc_input(self):
-        def check(fast_mode):
-            def fn(sparse_csc):
-                return torch.clone(sparse_csc).to_dense()
-
-            gradcheck(fn, torch.rand(2, 2, dtype=torch.double).to_sparse_csc().requires_grad_(True), check_sparse_nnz=True,
-                      check_batched_grad=False, fast_mode=fast_mode)
-
-            with self.assertRaisesRegex(RuntimeError, 'gradcheck expects all tensor inputs are dense'):
-                gradcheck(fn, torch.rand(2, 2, dtype=torch.double).to_sparse_csc().requires_grad_(True), check_sparse_nnz=False,
-                          check_batched_grad=False, fast_mode=fast_mode, masked=True)
-        check(fast_mode=True)
-        check(fast_mode=False)
-
-    def test_gradcheck_sparse_bsr_input(self):
-        def check(fast_mode):
-            def fn(sparse_bsr):
-                return torch.clone(sparse_bsr).to_dense()
-
-            gradcheck(fn, torch.rand(4, 8, dtype=torch.double).to_sparse_bsr((2, 2)).requires_grad_(True),
-                      check_sparse_nnz=True, check_batched_grad=False, fast_mode=fast_mode)
-
-            with self.assertRaisesRegex(RuntimeError, 'gradcheck expects all tensor inputs are dense'):
-                gradcheck(fn, torch.rand(2, 2, dtype=torch.double).to_sparse_bsr((2, 2)).requires_grad_(True),
-                          check_sparse_nnz=False, check_batched_grad=False, fast_mode=fast_mode, masked=True)
-        check(fast_mode=True)
-        check(fast_mode=False)
+    @parametrize('layout', (torch.sparse_coo, torch.sparse_csr, torch.sparse_csc, torch.sparse_bsr, torch.sparse_bsc))
```

Comment:
nice clean up

Reply:
yep, this is thanks to `to_sparse` supporting the `layout` and `blocksize` kw arguments.